drivers/overlay/overlay.go

```diff
@@ -692,8 +692,11 @@ func supportsOverlay(home string, homeMagic graphdriver.FsMagic, rootUID, rootGI
 
 	selinuxLabelTest := selinux.PrivContainerMountLabel()
 
-	if err := exec.Command("modprobe", "overlay").Run(); err != nil {
-		logrus.Warnf("Execution of `modprobe overlay` ended with error: %v", err)
+	// don't try to modprobe overlay if it's already loaded
+	if err := fileutils.Exists("/sys/module/overlay"); err != nil {
+		if err := exec.Command("modprobe", "overlay").Run(); err != nil {
```

Comment:
Do we actually have to load the module via modprobe? Is this not auto loaded when mounting the fs?

Reply:
the auto loading in the kernel seems to use a hardcoded path for modprobe:

```
$ grep CONFIG_MODPROBE_PATH /boot/config-$(uname -r)
CONFIG_MODPROBE_PATH="/usr/sbin/modprobe"
```

so if modprobe is installed at a different directory it would not be found.  I am fine with such a change because anyway the user can still load the module manually without us having to run modprobe each time we initialize the storage